$NetBSD: patch-5613,v 1.1 2019/07/17 01:44:31 manu Exp $

The scheduler now uses both the group's membership list as well as the
various OS-specific membership functions to determine whether a user belongs
to a named group (Issue #5613)

From upstream
https://github.com/apple/cups/commit/3c27d2a6ddf50a4cb02c0b7a464eaf7b6f1ea601

--- ./scheduler/auth.c
+++ ./scheduler/auth.c
@@ -1166,7 +1166,23 @@ cupsdCheckGroup(
 
     groupid = group->gr_gid;
 
+    for (i = 0; group->gr_mem[i]; i ++)
+    {
+     /*
+      * User appears in the group membership...
+      */
+
+      if (!_cups_strcasecmp(username, group->gr_mem[i]))
+	return (1);
+    }
+
 #ifdef HAVE_GETGROUPLIST
+   /*
+    * If the user isn't in the group membership list, try the results from
+    * getgrouplist() which is supposed to return the full list of groups a user
+    * belongs to...
+    */
+
     if (user)
     {
       int	ngroups;		/* Number of groups */
@@ -1187,13 +1203,6 @@ cupsdCheckGroup(
         if ((int)groupid == (int)groups[i])
 	  return (1);
     }
-
-#else
-    for (i = 0; group->gr_mem[i]; i ++)
-    {
-      if (!_cups_strcasecmp(username, group->gr_mem[i]))
-	return (1);
-    }
 #endif /* HAVE_GETGROUPLIST */
   }
   else
